src/NuGet.Clients/NuGet.CommandLine/Commands/SignCommand.cs

```diff
@@ -114,7 +114,7 @@ private HashAlgorithmName ValidateAndParseHashAlgorithm(string value, string nam
 
             if (!string.IsNullOrEmpty(value))
             {
-                if (!spec.AllowedHashAlgorithms.Contains(value, StringComparer.OrdinalIgnoreCase))
+                if (spec.AllowedHashAlgorithms.Contains(value, StringComparer.OrdinalIgnoreCase))
```

Comment:
@zhili1208 is fixing this in https://github.com/NuGet/NuGet.Client/pull/1845.

Please coordinate with him.